src/System.Security.AccessControl/src/System/Security/AccessControl/ACE.cs

```diff
@@ -149,7 +149,7 @@ internal void MarshalHeader(byte[] binaryForm, int offset)
 
             binaryForm[offset + 0] = (byte)AceType;
             binaryForm[offset + 1] = (byte)AceFlags;
-            binaryForm[offset + 2] = (byte)(Length >> 0);
+            binaryForm[offset + 2] = unchecked((byte)(Length >> 0));
             binaryForm[offset + 3] = (byte)(Length >> 8);
```

Comment:
Correct, because the ACE size is stored as an unsigned 16-bit integer:
https://msdn.microsoft.com/en-us/library/windows/desktop/aa374919.aspx